backend/app/controllers/app.py

```diff
@@ -425,37 +425,31 @@ def home():
     return "Welcome to the Backend!"
 
 
+user_repo = UserRepository(table_name="users")
+
 @app.route("/signup", methods=["POST"])
 def signup():
     data = request.get_json()
     print("Received data:", data)
+
     firstname = data.get("firstname")
     lastname = data.get("lastname")
     email = data.get("email")
     password = data.get("password")
     confirm_password = data.get("confirmPassword")
-    print(email)
 
+    # Validate required fields and password confirmation
     if password != confirm_password:
-        return (
-            jsonify({"code": 2, "error": True, "message": "Passwords do not match"}),
-            400,
-        )
+        return jsonify({"code": 2, "error": True, "message": "Passwords do not match"}), 400
 
-    if (
-        not firstname
-        or not lastname
-        or not email
-        or not password
-        or not confirm_password
-    ):
-        return (
-            jsonify({"code": 2, "error": True, "message": "All fields are required"}),
-            400,
-        )
+    if not firstname or not lastname or not email or not password or not confirm_password:
+        return jsonify({"code": 2, "error": True, "message": "All fields are required"}), 400
 
     try:
-        response = register_user_interactor(firstname, lastname, email, password)
+
```

Comment:
Nice! Did you convert all scenarios into interactors?

Reply:
Yes I converted it into 3 different interactors one is signup(register), login and change password.